app/src/main/java/com/google/samples/apps/sunflower/GardenFragment.kt

```diff
@@ -55,5 +56,11 @@ class GardenFragment : Fragment() {
             if (result != null && result.isNotEmpty())
                 adapter.submitList(result)
         })
+
+        adapter.openPlant.observe(viewLifecycleOwner, Observer { plantId ->
+            val direction =
+                GardenFragmentDirections.ActionGardenFragmentToPlantDetailFragment(plantId.peek())
```

Comment:
Can you change these to use `plantId.consume()?.let { ... }`? We want to make sure that even if there are (somehow) multiple observers, only one handles the event. [Recommended: Use an Event wrapper](https://medium.com/androiddevelopers/livedata-with-snackbar-navigation-and-other-events-the-singleliveevent-case-ac2622673150#0e87)

Reply:
@nic0lette my mis, please check again.